### PR TITLE
Refactor PeepRideSubState to enum class fix #12390

### LINKED ISSUE #12390: Refactor PeepRideSubState to use strong enum

### PR DESCRIPTION
**Pull request by rpstester**
Also made a RideSubState as a union with SubState and SittingSubState
Updated all values across all files.

I have revised the prior pull request #12931 (now that I have learned more about Git).  Please let me know your feedback. Thanks!

### CODE REVIEW

**Review by tupaschoal — APPROVED** [review]
This looks fine to me. @duncanspumpkin said he'd like to take it for a spin on the other PR, so I'll wait for his word before merging.

File: src/openrct2/peep/Guest.cpp [hunk 1]
```diff
@@ -3705,7 +3705,7 @@ void Guest::UpdateRideAdvanceThroughEntrance()
         return;
     }
 
-    Guard::Assert(SubState == PEEP_RIDE_LEAVE_ENTRANCE, "Peep substate should be LEAVE_ENTRANCE");
+    Guard::Assert(RideSubState == PeepRideSubState::LeaveEntrance, "Peep substate should be LEAVE_ENTRANCE");
```
Comment (duncanspumpkin) [inline src/openrct2/peep/Guest.cpp:3708]: Update the assert comment

**Review by duncanspumpkin — APPROVED** [review]

**Comment by tupaschoal** [comment]
You'll also need to rebase to solve the merge conflicts

**Comment by rpstester** [comment]
@tupaschoal Rebase with conflict resolution complete!

File: src/openrct2/peep/Guest.cpp (hunk 1 above)
Comment (tupaschoal) [inline src/openrct2/peep/Guest.cpp:3708]:
```suggestion
    Guard::Assert(RideSubState == PeepRideSubState::LeaveEntrance, "Peep ride substate should be LeaveEntrance");
```

**Review by tupaschoal — COMMENTED** [review]
Looks good, just need to apply Duncan's suggestion

File: src/openrct2/peep/Guest.cpp (hunk 1 above)
Comment (rpstester) [reply src/openrct2/peep/Guest.cpp]: OK, I see what you mean. It is done!

**Pull request merged**